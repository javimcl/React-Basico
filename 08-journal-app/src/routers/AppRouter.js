import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    Redirect
} from "react-router-dom";
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route path="/auth"
                        component={AuthRouter}>

                    </Route>
                    <Route path="/"
                        component={JournalScreen}>

                    </Route>
                    <Redirect to="/auth/login"/>
                </Switch>

            </div>
        </Router>
    );
};
