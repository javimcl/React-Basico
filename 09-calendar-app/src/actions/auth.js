import { documentNotReadyOrSSRTesting } from "react-modal/lib/helpers/ariaAppHider";
import Swal from "sweetalert2";
import { fetchSinToken } from "../helpers/fetch";
import { types } from "../types/types";


export const startLogin = (email, password) => {
    console.log(email, password);
    
    return async ( dispatch) => {
     
        const resp = await fetchSinToken('auth', {email, password}, 'POST');
        const body = await resp.json();
        
        if (body.ok) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());

            dispatch(login({
                uid: body.uid,
                name:body.name
            }))
        } else {
            Swal.fire('Error', body.msg, 'error')
        }
    }
}


const login = (user) => ({
    type: types.authLogin,
    payload: user
})