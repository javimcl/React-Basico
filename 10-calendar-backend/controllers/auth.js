const bcrypt = require('bcryptjs');
const { response } = require('express');
const { validationResult } = require('express-validator');
const { generarJWT } = require('../helpers/jwt');
const Usuario = require('../models/Usuario');


const crearUsuario = async (req, res = response) => {

    const { email, password } = req.body;

    try {

        let usuario = await Usuario.findOne({ email });

        if (usuario) {
            return res.status(400).json({
                ok: false,
                msg: 'Un usuario ya existe con ese correo'
            })

        }

        usuario = new Usuario(req.body);

        //Encriptar Contrasenia
        const salt = bcrypt.genSaltSync();
        usuario.password = bcrypt.hashSync(password, salt);

        await usuario.save();

        //Generar JWT
        const token = await generarJWT(usuario.id, usuario.name);

        res.status(201).json({
            ok: true,
            uid: usuario.id,
            name: usuario.name,
            token

        })
    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            msg: 'Por favor comuniquese con el administrador',

        })


    }

}

const loginUsuario = async (req, res = response) => {

    const { email, password } = req.body;

    try {
        const usuario = await Usuario.findOne({ email });

        if (!usuario) {
            return res.status(400).json({
                ok: false,
                msg: 'El usuario no existe con ese mail'
            })

        }

        //Confirmar los passwords
        const validaPassword = bcrypt.compareSync(password, usuario.password);
        if (!validaPassword) {
            return res.status(400).json({
                ok: false,
                msg: 'Password incorrecto'
            })
        }

        //Generar JWT
        const token = await generarJWT(usuario.id, usuario.name);


        // Generar nuestro JWT
        res.status(201).json({
            ok: true,
            uid: usuario.id,
            name: usuario.name,
            token
        })

    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            msg: 'Por favor comuniquese con el administrador',

        })
    }




}

const revalidarToken = (req, res = response) => {

    res.json({
        ok: true,
        msg: 'renew'
    })
}

module.exports = {
    crearUsuario,
    loginUsuario,
    revalidarToken
}