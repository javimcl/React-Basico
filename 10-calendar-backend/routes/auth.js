/*
    Rutas de Usuarios / Auth
    host + /api/auth

*/

const {Router} = require('express');
const { check } = require('express-validator');
const router = Router();

const {crearUsuario, loginUsuario, revalidarToken} = require('../controllers/auth');

router.post(
    '/new', 
    [ //middleware
        check('name', 'El nombre es obligatorio').not().isEmpty()
    ]
     ,
    crearUsuario);

router.post('/',loginUsuario);

router.get('/renew', revalidarToken);

module.exports = router