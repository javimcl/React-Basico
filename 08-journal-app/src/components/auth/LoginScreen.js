import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { login } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {

  const dispatch = useDispatch();

  const [formValues, handleInputChange] = useForm({
    email: 'javier',
    password: '1234'
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(login(123,'Javier'))

  }


  return (
    <>
      <h3 className="auth__title">Login</h3>
      <form onSubmit={handleLogin}>
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          autoComplete='off'
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
          value={password}
          onChange={handleInputChange}
        />
        <button className='btn btn-primary btn-block'
          type="submit"
        >
          Login
        </button>

        <div className='auth__social-networks'>
          <p>Login with social metworks</p><div
            className="google-btn"
          >
            <div className="google-icon-wrapper">
              <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className='link'
          to="/auth/register">
          Create new account
        </Link>

      </form>
    </>
  );
};
