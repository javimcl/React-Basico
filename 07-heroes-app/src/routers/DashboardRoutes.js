import React from 'react';
import { Routes, Route, Link, BrowserRouter } from "react-router-dom";
import { Navbar } from '../components/ui/Navbar';
import { DcScreen } from '../components/dc/DcScreen';
import { LoginScreen } from '../components/login/LoginScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <Routes>
                
                <Route path="marvel" element={<MarvelScreen />} />
                <Route path="dc" element={<DcScreen />} />
                
                <Route path="search" element={<SearchScreen />} />

                <Route path="hero" element={<HeroScreen />} />
                
                <Route path="/login" element={<LoginScreen />} />

            </Routes>
        </>
    );
};
