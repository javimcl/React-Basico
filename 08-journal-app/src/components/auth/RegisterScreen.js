import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';

import { setError, removeError } from '../../actions/ui'
import { useDispatch, useSelector } from 'react-redux';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';

export const RegisterScreen = () => {

  const dispatch = useDispatch();

  const { msgError } = useSelector(state => state.ui);

 // console.log(msgError);

  const initial = {
    name: 'Hernando',
    email: 'nando@iess.gob.ec',
    password: '123456',
    password2: '123456'
  }

  const [formValues, handleInputChange] = useForm(initial);

  const { name, email, password, password2 } = formValues;

  const handleRegister = (e) => {
    e.preventDefault();    
    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, name))
    }
  }

  const isFormValid = () => {

    if (name.trim().length === 0) {
      dispatch(setError('Name is required'));
      return false;
    } else if (!validator.isEmail(email)) {      
      dispatch(setError('Email is not valida'));
      return false;

    } else if (password !== password2 || password.length < 5) {
      dispatch(setError('Password should be at least 6 characters and match each other'))
      return false;
    }

    dispatch(removeError());


    return true;
  }


  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form onSubmit={handleRegister}
        className='animate__animated animate__fadeIn animate__faster'>
        {
          msgError &&
          (
            <div className='auth__alert-error'>
              {msgError}
            </div>
          )
        }
        <input
          type="text"
          placeholder="Name"
          name="name"
          className="auth__input"
          autoComplete='off'
          onChange={handleInputChange}
          value={name}
        />
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          autoComplete='off'
          onChange={handleInputChange}
          value={email}
        />
        <input
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
          onChange={handleInputChange}
          value={password}
        />
        <input
          type="password"
          placeholder="Confirm Password"
          name="password2"
          className="auth__input"
          onChange={handleInputChange}
          value={password2}
        />
        <button className='btn btn-primary btn-block mb-5'
          type="submit"
          onClick={handleRegister}
        >
          Register
        </button>

        <div className='auth__social-networks'>
          <p>Login with social metworks</p><div
            className="google-btn"
          >
            <div className="google-icon-wrapper">
              <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className='link'
          to="/auth/login">
          Already registered?
        </Link>

      </form>
    </>
  );
};
