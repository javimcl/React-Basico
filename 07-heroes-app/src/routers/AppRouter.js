import React from 'react';
import { Routes, Route, Link, BrowserRouter } from "react-router-dom";
import { DcScreen } from '../components/dc/DcScreen';
import { LoginScreen } from '../components/login/LoginScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/Navbar';
import { PrivateRoute } from '../hooks/PrivateRoute';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/login" element={<LoginScreen />} />

        <Route path="/*" element={
          <PrivateRoute>
            <DashboardRoutes />
          </PrivateRoute>

        } />




        {/* <Route path="/*" element={<DashboardRoutes/>} /> */}

      </Routes>
    </BrowserRouter>
  );
};
