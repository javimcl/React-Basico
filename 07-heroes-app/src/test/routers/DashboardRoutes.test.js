import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

describe('PRuebas en <Dashboard/>', () => {

    const contextValue = {
        user: {
            logged: true,
            name: 'Javier'
        }
    }

    test('debe de mostrarse correctamente - Marvel', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/']}>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Javier')
    });

    test('debe de mostrarse correctamente - Marvel', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/dc']}>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('DC screen')
    });

    
});
