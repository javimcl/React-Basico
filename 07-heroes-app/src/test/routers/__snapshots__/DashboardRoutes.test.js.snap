// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRuebas en <Dashboard/> debe de mostrarse correctamente - Marvel 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Javier
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  />,
]
`;

exports[`PRuebas en <Dashboard/> debe de mostrarse correctamente - Marvel 2`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-linkactive"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Javier
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        DC screen
      </h1>
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Batman"
                  className="card-img-top"
                  src="/assets/dc-batman.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Batman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    href="/hero/dc-batman"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Superman"
                  className="card-img-top"
                  src="/assets/dc-superman.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Superman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Kal-El
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Action Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-superman"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Flash"
                  className="card-img-top"
                  src="/assets/dc-flash.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Flash
                  </h5>
                  <p
                    className="card-text"
                  >
                    Jay Garrick
                  </p>
                  <p
                    className="text-muted"
                  >
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Flash Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-flash"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Green Lantern"
                  className="card-img-top"
                  src="/assets/dc-green.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Green Lantern
                  </h5>
                  <p
                    className="card-text"
                  >
                    Alan Scott
                  </p>
                  <p
                    className="text-muted"
                  >
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      All-American Comics #16
                    </small>
                  </p>
                  <a
                    href="/hero/dc-green"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Green Arrow"
                  className="card-img-top"
                  src="/assets/dc-arrow.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Green Arrow
                  </h5>
                  <p
                    className="card-text"
                  >
                    Oliver Queen
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      More Fun Comics #73
                    </small>
                  </p>
                  <a
                    href="/hero/dc-arrow"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Wonder Woman"
                  className="card-img-top"
                  src="/assets/dc-wonder.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wonder Woman
                  </h5>
                  <p
                    className="card-text"
                  >
                    Princess Diana
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      All Star Comics #8
                    </small>
                  </p>
                  <a
                    href="/hero/dc-wonder"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Martian Manhunter"
                  className="card-img-top"
                  src="/assets/dc-martian.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Martian Manhunter
                  </h5>
                  <p
                    className="card-text"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    className="text-muted"
                  >
                    Martian Manhunter
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #225
                    </small>
                  </p>
                  <a
                    href="/hero/dc-martian"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Robin/Nightwing"
                  className="card-img-top"
                  src="/assets/dc-robin.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Robin/Nightwing
                  </h5>
                  <p
                    className="card-text"
                  >
                    Dick Grayson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #38
                    </small>
                  </p>
                  <a
                    href="/hero/dc-robin"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Blue Beetle"
                  className="card-img-top"
                  src="/assets/dc-blue.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Blue Beetle
                  </h5>
                  <p
                    className="card-text"
                  >
                    Dan Garret
                  </p>
                  <p
                    className="text-muted"
                  >
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Mystery Men Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-blue"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Black Canary"
                  className="card-img-top"
                  src="/assets/dc-black.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Black Canary
                  </h5>
                  <p
                    className="card-text"
                  >
                    Dinah Drake
                  </p>
                  <p
                    className="text-muted"
                  >
                    Dinah Drake, Dinah Lance
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Flash Comics #86
                    </small>
                  </p>
                  <a
                    href="/hero/dc-black"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
